sound/soc/intel/boards/sof_rt5682.c

```diff
@@ -66,12 +66,29 @@ struct sof_card_private {
 	bool common_hdmi_codec_drv;
 };
 
+const char * sof_override_tplg_name=NULL;
+
 static int sof_rt5682_quirk_cb(const struct dmi_system_id *id)
 {
 	sof_rt5682_quirk = (unsigned long)id->driver_data;
 	return 1;
 }
-
+static int sof_rt5682_tplg_cb(const struct dmi_system_id *id)
+{
+	sof_override_tplg_name = (const char *)id->driver_data;
+	return 1;
+}
+static const struct dmi_system_id sof_rt5682_tplg_table[] = {
+	{
+		.callback = sof_rt5682_tplg_cb,
+		.matches = {
+			/* will replace with real product name later*/
+			DMI_MATCH(DMI_PRODUCT_NAME, "Google Unique Product Name"),
+		},
+		.driver_data = (void *)("sof-tgl-max98373-rt5682-ampI2S2.tplg"),
```

Comment:
nitpick: the string is of type `char *` so, no type-casting should be needed. In fact maybe you could drop this static struct altogether - see below.

Reply:
> nitpick: the string is of type `char *` so, no type-casting should be needed. In fact maybe you could drop this static struct altogether - see below.

sure, will fix this when posting. Thanks!